ygen/directory.go

```diff
@@ -276,6 +276,17 @@ func getOrderedDirDetails(langMapper LangMapper, directory map[string]*Directory
 			if hasShadowField {
 				nd.YANGDetails.ShadowSchemaPath = util.SchemaTreePathNoModule(shadowField)
 			}
+			// If IROptions.PathOriginName has a value, the value is set to the Origin of the node.
+			// Else if IROptions.UseModuleNameAsPathOrigin of the node is true,
+			// YANG module name is set to the Origin of the node.
+			// Else "openconfig" is set to the Origin of the node.
+			if opts.PathOriginName != "" {
+				nd.YANGDetails.Origin = opts.PathOriginName
+			} else if opts.UseModuleNameAsPathOrigin {
+				nd.YANGDetails.Origin = nd.YANGDetails.BelongingModule
+			} else {
+				nd.YANGDetails.Origin = "openconfig"
+			}
```

Comment:
```suggestion
            switch {
            case opts.PathOriginName != "":
				nd.YANGDetails.Origin = opts.PathOriginName
			case opts.UseModuleNameAsPathOrigin:
				nd.YANGDetails.Origin = nd.YANGDetails.BelongingModule
			default:
			    // TODO: read the origin from the relevant YANG extension.
				nd.YANGDetails.Origin = ""
			}
```

Two changes:
1. Use a switch rather than if/else if/else to make this easier to extend in the future.
2. Switch the "default" origin to be "", since making it "openconfig" ties this statically to OpenConfig models, which might not be the case.
3. Add a TODO to say that we should be able to read this value from the relevant extension in the YANG model (https://github.com/openconfig/public/blob/master/release/models/openconfig-extensions.yang#L219-L228).

Reply:
@robshakir 
Fixed as suggeted.